modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/services/rest/TestrayComponent.ts

```diff
@@ -28,8 +29,17 @@ const adapter = ({
 	r_teamToComponents_c_teamId,
 });
 
-const createComponent = (component: Component) =>
-	fetcher.post('/components', adapter(component));
+const createComponent = async (component: Component) => {
+	const response = await fetcher(
+		`/components?filter=${searchUtil.eq('name', component.name)}`
+	);
+
+	if (response.items?.length) {
+		throw new Error();
```

Comment:
@samuelleolopes you need to add the error message inside `Error();`

Reply:
@samuelleolopes , it's possible to create a component with the same name, since they belong to a different project